spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/logging/it/StackdriverLoggingIntegrationTests.java

```diff
@@ -0,0 +1,124 @@
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.cloud.gcp.autoconfigure.sql.GcpCloudSqlAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.storage.GcpStorageAutoConfiguration;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+/**
+ * @author João André Martins
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@RunWith(SpringRunner.class)
+public class StackdriverLoggingIntegrationTests {
+
+	private static final Log LOGGER = LogFactory.getLog(StackdriverLoggingIntegrationTests.class);
+
+	@Autowired
+	private GcpProjectIdProvider projectIdProvider;
+
+	@Autowired
+	private CredentialsProvider credentialsProvider;
+
+	@LocalServerPort
+	private int port;
+
+	private static final LocalDateTime NOW = LocalDateTime.now();
+
+	@BeforeClass
+	public static void enableTests() {
+		assumeThat(System.getProperty("it.logging")).isEqualTo("true");
+	}
+
+	@Test
+	public void test() throws InterruptedException, IOException {
+		URL url = new URL("http://localhost:" + this.port);
+		HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+		connection.setRequestProperty("x-cloud-trace-context", "everything-zen");
+		connection.getResponseMessage();
+		connection.disconnect();
+
+		CredentialsProvider credentialsProvider = this.credentialsProvider;
+		Logging logClient = LoggingOptions.newBuilder()
+				.setCredentials(credentialsProvider.getCredentials())
+				.build().getService();
+		GcpProjectIdProvider projectIdProvider = this.projectIdProvider;
+		Page<LogEntry> page;
+		int pageSize = 0;
+		int counter = 0;
+		do {
+			Thread.sleep(100);
+			page = logClient.listLogEntries(
+					Logging.EntryListOption.filter("textPayload:\"#$%^&" + NOW + "\" AND"
+							+ " logName=\"projects/" + projectIdProvider.getProjectId()
+							+ "/logs/spring.log\""));
+			for (LogEntry entry : page.getValues()) {
+				pageSize++;
+			}
+		} while (pageSize == 0 && counter++ < 2);
+		assertThat(pageSize).isEqualTo(1);
+		assertThat(page.getValues().iterator().next().getLabels().get("trace_id"))
+				.isEqualTo("everything-zen");
+	}
+
+	@RestController
+	@SpringBootApplication(exclude = {
+			GcpCloudSqlAutoConfiguration.class,
+			GcpStorageAutoConfiguration.class,
+			DataSourceAutoConfiguration.class
+	})
+	static class LoggingApplication {
+
+		public static void main(String[] args) {
+			SpringApplication.run(LoggingApplication.class, args);
+		}
+
+		@GetMapping
+		public void log() {
```

Comment:
Well, since it's `void` it sounds more like `HEAD`, but not `GET`: some inconsistency in REST protocol usage: https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html :smile: